agent/src/config.cc

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2024 Centreon
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ */
+
+#include <rapidjson/document.h>
+#include <re2/re2.h>
+
+#include "com/centreon/common/rapidjson_helper.hh"
+#include "com/centreon/exceptions/msg_fmt.hh"
+#include "config.hh"
+
+using namespace com::centreon::agent;
+using com::centreon::common::rapidjson_helper;
+
+static constexpr std::string_view _config_schema(R"(
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "agent config",
+    "properties": {
+        "host": {
+            "description": "name of host configured in centreon, if not given, hostname will be used",
+            "type": "string",
+            "minLength": 5
+        },
+        "endpoint": {
+            "description": "endpoint of poller where agent has to connect or listening endpoint in case of reverse_connection",
```

Comment:
```suggestion
            "description": "Endpoint where agent has to connect to on the poller side or listening endpoint on the agent side in case of reverse_connection",
```